Unable to build pregen files

### Reproduction steps

1. cd ~/connectedhomeip
2. export ZAP_INSTALL_PATH=~/Downloads/zap-mac/
3. scripts/codepregen.py ${OUTPUT_DIRECTORY:-./zzz_pregenerated/}
4. export PKG_CONFIG_PATH="/opt/homebrew/opt/openssl@3/lib/pkgconfig"
5. source scripts/bootstrap.sh
6. source scripts/activate.sh
7. cd ~/tools
8. git clone https://github.com/espressif/esp-idf.git
9. cd esp-idf
10. git pull
11. git checkout v4.4.4
12. git submodule update --init
13. ./install.sh
14. . ./export.sh
15. cd ~/connectedhomeip/examples/all-clusters-app/esp32
16. export ZAP_INSTALL_PATH=~/Downloads/zap-mac/
17. idf.py -DCHIP_CODEGEN_PREGEN_DIR=~/connectedhomeip/zzz_pregenerated/ build

Currently, I am getting the following error;

```
[611/1215] Performing build step for 'chip_gn'
FAILED: esp-idf/chip/chip_gn-prefix/src/chip_gn-stamp/chip_gn-build esp-idf/chip/lib/libCHIP.a /Users/priyankahonnakasturi/connectedhomeip/examples/all-clusters-app/esp32/build/esp-idf/chip/chip_gn-prefix/src/chip_gn-stamp/chip_gn-build /Users/priyankahonnakasturi/connectedhomeip/examples/all-clusters-app/esp32/build/esp-idf/chip/lib/libCHIP.a 
cd /Users/priyankahonnakasturi/connectedhomeip/examples/all-clusters-app/esp32/build/esp-idf/chip && ninja esp32
[0/1] Regenerating ninja files
ERROR at build arg file (use "gn args <out_dir>" to edit):14:37: Build argument has no effect.
chip_code_pre_generated_directory = "/Users/priyankahonnakasturi/connectedhomeip/zzz_pregenerated/"
                  ^--------------------------------------------------------------
The variable "chip_code_pre_generated_directory" was set as a build argument
but never appeared in a declare_args() block in any buildfile.

To view all possible args, run "gn args --list <out_dir>"
FAILED: build.ninja.stamp 
../../../../../../.environment/cipd/packages/pigweed/gn --root=../../../../../../config/esp32 -q --check --fail-on-unused-args --regeneration gen .
ninja: error: rebuilding 'build.ninja': subcommand failed
[613/1215] Building C object esp-...nghttp2/lib/nghttp2_session.c.obj
ninja: build stopped: subcommand failed.
ninja failed with exit code 1
esp32 %
```

Expected result is that we must be able to build zzz_pregenerated  via the command   
`idf.py -DCHIP_CODEGEN_PREGEN_DIR=~/connectedhomeip/zzz_pregenerated/ build`

So that we can modify the endpoints from ~/connectedhomeip/zzz_pregenerated/examples/all-clusters-app/all-clusters-common/all-clusters-app/zap/app-templates/zap-generated/endpoint_config.h

and be able to flash M5 board with the respective endpoints.

### Bug prevalence

always

### GitHub hash of the SDK that was being used

fcf37c239c5c9858d312030a4549dbe91cee8e2b

### Platform

darwin

### Platform Version(s)

_No response_

### Anything else?

_No response_

--- a/config/esp32/components/chip/CMakeLists.txt
+++ b/config/esp32/components/chip/CMakeLists.txt
@@ -309,11 +309,21 @@ if(CONFIG_ENABLE_PW_RPC)
     list(APPEND chip_libraries "${CMAKE_CURRENT_BINARY_DIR}/lib/libPwRpc.a")
 endif()
 
+# When using the pregenerated files, there is a edge case where an error appears for
+# undeclared argument chip_code_pre_generated_directory. To get around with it we are
+# disabling the --fail-on-unused-args flag.
+# For more, see: https://github.com/project-chip/connectedhomeip/issues/27636
+if (CHIP_CODEGEN_PREGEN_DIR)
+    set(GN_CONFIGURE_COMMAND ${GN_EXECUTABLE} --root=${GN_ROOT_TARGET} gen --check ${CMAKE_CURRENT_BINARY_DIR})
+else ()
+    set(GN_CONFIGURE_COMMAND ${GN_EXECUTABLE} --root=${GN_ROOT_TARGET} gen --check --fail-on-unused-args ${CMAKE_CURRENT_BINARY_DIR})
+endif ()
+
 externalproject_add(
     chip_gn
     SOURCE_DIR              ${CHIP_ROOT}
     BINARY_DIR              ${CMAKE_CURRENT_BINARY_DIR}
-    CONFIGURE_COMMAND       ${GN_EXECUTABLE} --root=${GN_ROOT_TARGET} gen --check --fail-on-unused-args ${CMAKE_CURRENT_BINARY_DIR}
+    CONFIGURE_COMMAND       ${GN_CONFIGURE_COMMAND}
     BUILD_COMMAND           ninja "esp32"
     INSTALL_COMMAND         ""
     BUILD_BYPRODUCTS        ${chip_libraries}

